docs/source/advanced/mixed_precision.rst

```diff
@@ -5,49 +5,130 @@
-========================
-
-Mixed precision combines the use of both FP32 and lower bit floating points (such as FP16) to reduce memory footprint during model training, resulting in improved performance.
 
-Lightning offers mixed precision training for GPUs and CPUs, as well as bfloat16 mixed precision training for TPUs.
+#########
+Precision
+#########
+
+There are numerous benefits to using numerical formats with lower precision than the 32-bit floating-point or higher precision such as 64-bit floating-point.
+
+Lower precision, such as the 16-bit floating-point, requires less memory, enabling the training and deployment of larger neural networks. Second, they require less memory bandwidth, thereby speeding up data transfer operations. Third, math operations run much faster in reduced precision, especially on GPUs with Tensor Core support for that precision.
+
+Higher precision, such as the 64-bit floating-point, offers <TODO add something>
+
+Following are the precisions available in Lightning along with their supported Accelerator:
+
+.. list-table:: Precision with Accelerators
+   :widths: 20 20 20 20 20
+   :header-rows: 1
+
+   * - Precision
+     - CPU
+     - GPU
+     - TPU
+     - IPU
+   * - 16
+     - No
+     - Yes
+     - No
+     - Yes
+   * - BFloat16
+     - Yes
+     - Yes
+     - Yes
+     - No
+   * - 32
+     - Yes
+     - Yes
+     - Yes
+     - Yes
+   * - 64
+     - Yes
+     - Yes
+     - No
+     - No
+
+
+***************
+Mixed Precision
+***************
+
+Most deep learning frameworks, including PyTorch, train with 32-bit floating-point (FP32) arithmetic by default. However, this is not essential for many deep learning models to achieve complete accuracy. Mixed precision training offers significant computational speedup
+by performing operations in half-precision format while storing minimal information in single-precision to retain as much information as possible in critical parts of the network. Since the introduction of Tensor Cores in the Volta and Turing architectures, significant
+training speedups are experienced by switching to mixed precision. It combines both FP32 and lower bit floating-points (such as FP16) to reduce memory footprint during model
+training/evaluation, resulting in improved performance. It does so by identifying the steps that require full precision and using a 32-bit floating-point for only those steps while using a 16-bit floating-point everywhere else. Mixed precision training achieves all these benefits
+while ensuring that no task-specific accuracy is lost compared to full precision training.
 
 .. note::
 
-    In some cases it is important to remain in FP32 for numerical stability, so keep this in mind when using mixed precision.
+    In some cases, it is essential to remain in FP32 for numerical stability, so keep this in mind when using mixed precision.
+    For example, when running scatter operations during the forward (such as torchpoint3d), computation must remain in FP32.
```

Comment:
I think it would be worth to add examples where a user could cast its loss or softmax to FP32 using autocast while training in precision=16, which we know to bring unstability.

Reply:
do you have a direct example from torchpoint3d?

Reply:
added.